x-pack/plugins/cases/public/components/all_cases/selector_modal/all_cases_selector_modal.test.tsx

```diff
@@ -6,62 +6,88 @@
  */
 
 import React from 'react';
-import { mount } from 'enzyme';
 
 import { AllCasesSelectorModal } from '.';
-import { TestProviders } from '../../../common/mock';
-import { AllCasesList } from '../all_cases_list';
+import { AppMockRenderer, createAppMockRenderer } from '../../../common/mock';
+import userEvent from '@testing-library/user-event';
+import { act, waitFor } from '@testing-library/react';
 
-jest.mock('../all_cases_list', () => ({ AllCasesList: jest.fn().mockReturnValue(<></>) }));
+jest.mock('../../../containers/api');
+jest.mock('../../../containers/user_profiles/api');
 
 const onRowClick = jest.fn();
 const defaultProps = {
   onRowClick,
 };
 
 describe('AllCasesSelectorModal', () => {
+  let appMockRenderer: AppMockRenderer;
+
   beforeEach(() => {
     jest.clearAllMocks();
+    appMockRenderer = createAppMockRenderer();
   });
 
   it('renders', () => {
-    const wrapper = mount(
-      <TestProviders>
-        <AllCasesSelectorModal {...defaultProps} />
-      </TestProviders>
-    );
+    const res = appMockRenderer.render(<AllCasesSelectorModal {...defaultProps} />);
+
+    expect(res.getByTestId('all-cases-modal')).toBeInTheDocument();
+  });
+
+  it('Closing modal when pressing the x icon', () => {
+    const res = appMockRenderer.render(<AllCasesSelectorModal {...defaultProps} />);
+
+    act(() => {
+      userEvent.click(res.getByLabelText('Closes this modal window'));
+    });
+
+    expect(res.queryByTestId('all-cases-modal')).toBeFalsy();
+  });
+
+  it('Closing modal when pressing the cancel button', () => {
+    const res = appMockRenderer.render(<AllCasesSelectorModal {...defaultProps} />);
 
-    expect(wrapper.find(`[data-test-subj='all-cases-modal']`).exists()).toBeTruthy();
+    act(() => {
+      userEvent.click(res.getByTestId('all-cases-modal-cancel-button'));
+    });
+
+    expect(res.queryByTestId('all-cases-modal')).toBeFalsy();
+  });
+
+  it('should hide bulk actions and row actions', async () => {
```

Comment:
Is it hiding them by default because nothing was selected? If so, could we update the name to say why they are hidden, maybe something like:

`should hide bulk actions and row actions when no rows are selected`

Might be nice to have for the tests below too.

Reply:
Bulk actions and row actions should not be shown in the modal (on selector view). The user should not perform bulk actions when selecting a case from a modal. I changed the title to make it more clear.